File: old/launchpad_50_easy_questions/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> path;
        vector<int> result;
        explore(nums, target, 0, path, result);
        return result;
    }

private:
    void explore(vector<int>& nums, int target, int start, vector<int>& path, vector<int>& result) {
        if (path.size() == 2) {
            int sum = nums[path[0]] + nums[path[1]];
            if (sum == target) {
                result = path;
            }
            return;
        }

        for (int i = start; i < nums.size(); i++) {
            if (!result.empty()) return;

            path.push_back(i);
            explore(nums, target, i + 1, path, result);
            path.pop_back();
        }
    }
};

File: old/launchpad_50_easy_questions/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        current_items = []
        solution_items = []
        solution_items = self.explore(0, target, current_items, solution_items, nums)
        return solution_items

    def explore(self, start, target, current_items, solution_items, nums):
        if len(current_items) == 2:
            if nums[current_items[0]] + nums[current_items[1]] == target:
                solution_items = current_items[:]
                return solution_items
            return []

        for i in range(start, len(nums)):
            current_items.append(i)
            solution_items = self.explore(
                i + 1, target, current_items, solution_items, nums
            )
            current_items.pop()

            if solution_items:
                return solution_items

        return []
